src/main/java/com/edmunds/tools/databricks/maven/UpsertClusterMojo.java

```diff
@@ -0,0 +1,128 @@
+ */
+@Mojo(name = "upsert-cluster")
+public class UpsertClusterMojo extends BaseDatabricksMojo {
+
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMapperUtils.getObjectMapper();
+    private static final Map<String, String> SPARK_ENV_VARS = new HashMap<String, String>() {{
+        put("PYSPARK_PYTHON", "/databricks/python3/bin/python3");
+    }};
+
+    /**
+     * The databricks cluster json file that contains all of the information for how to create databricks cluster.
+     */
+    @Parameter(defaultValue = "${project.build.resources[0].directory}/databricks-plugin/databricks-cluster-settings.json", property = "dbClusterFile")
+    protected File dbClusterFile;
+
+    /**
+     * If true, command will fail if cluster with specified name already exists.
+     */
+    @Parameter(property = "failOnClusterExists")
+    protected boolean failOnClusterExists = true;
+
+    public void execute() throws MojoExecutionException {
+        ClusterTemplateModel[] cts;
+        try {
+            cts = OBJECT_MAPPER.readValue(dbClusterFile, ClusterTemplateModel[].class);
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to parse config", e);
+        }
+
+        for (ClusterTemplateModel ct : cts) {
+            ClusterService clusterService = getDatabricksServiceFactory().getClusterService();
+            String clusterId = convertClusterNamesToIds(clusterService, Collections.singletonList(ct.getClusterName()))
+                    .stream().findFirst().orElse(StringUtils.EMPTY);
+
+            CreateClusterRequest request = new CreateClusterRequest.CreateClusterRequestBuilder(
+                    ct.getNumWorkers(), ct.getClusterName(), ct.getSparkVersion(), ct.getNodeTypeId())
+                    .withAwsAttributes(ct.getAwsAttributes())
+                    .withAutoterminationMinutes(ct.getAutoterminationMinutes())
+                    .withSparkEnvVars(SPARK_ENV_VARS)
+                    .build();
+
+            String logMessage = String.format("Creating cluster: name=[%s]", ct.getClusterName());
+            try {
+                if (StringUtils.isNotEmpty(clusterId)) {
+                    logMessage = String.format("Updating cluster: name=[%s], id=[%s]", ct.getClusterName(), clusterId);
+                    if (failOnClusterExists) {
+                        throw new MojoExecutionException("Exception while " + logMessage + ". Cluster already exists");
+                    }
+                }
+                getLog().info(logMessage);
+                if (StringUtils.isNotEmpty(clusterId)) {
+                    //note that delete is an alias to terminate: https://docs.databricks.com/api/latest/clusters.html#delete-terminate
+                    clusterService.delete(clusterId);
+                }
+                clusterId = clusterService.create(request);
+
+                getLog().info(String.format("Attaching libraries to the cluster: name=[%s], id=[%s], libs=[%s]",
```

Comment:
I think it would be clearer, if library attachment happened in it's own function.